e2e-test/src/tests/delete.test.js

```diff
@@ -0,0 +1,66 @@
+import { vBtn } from 'jest-puppeteer-vuetify';
+import { CLIENT_URL } from '../util';
+
+
+describe('delete dataset', () => {
+  it('upload one dataset and then delete it', async () => {
+    await Promise.all([
+      page.goto(CLIENT_URL),
+      page.waitForNavigation({ waitUntil: 'networkidle0' }),
+    ]);
+    await expect(page).toClickXPath(vBtn('Your Datasets'));
+
+    const fileInputElementHandle = await page.waitForXPath("//input[@type='file']"); // $0.value == "" is true
+    await fileInputElementHandle.uploadFile('../e2e-test/data/Cancer chemo muscle (NMR).csv');
+    await page.waitForXPath("//div[@class='v-list__tile__title'][contains(text(),'Cancer_chemo_muscle_NMR.csv ')]");
+    await page.waitForXPath("//div[@class='v-list__tile__title'][contains(text(),'Cancer_chemo_muscle_NMR.csv ')]//following::span[contains(text(),'Dataset processed with 1 validation failures')]");
+
+    // click on 'X'
+    await page.waitForXPath("//div[@class='v-btn__content']//following::i[@class='v-icon mdi mdi-close theme--light']");
+    await expect(page).toClickXPath(vBtn("//i[@class='v-icon mdi mdi-close theme--light']"));
+
+    // make sure it is deleting one dataset
+    await page.waitForXPath("//div[@class='v-card__title headline'][contains(text(),'Really delete 1 dataset?')]");
+    await expect(page).toClickXPath(vBtn('Delete'));
+
+    // make sure it is deleted
+    await expect(page).not.toContainXPath("//div[@class='v-list__tile__title'][contains(text(),'Cancer_chemo_muscle_NMR.csv ')]");
+  });
+
+  it('upload 3 datasets and then clear all', async () => {
```

Comment:
The `upload 3 datasets` part isn't really what's being tested, it's just a setup step. I would just call this test `clear all` as that is what is being tested here.
Similar comments apply to other tests as well, I didn't notice earlier.